modules/custom/falcon_mail/src/Plugin/Mail/FalconMailSystem.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Drupal\falcon_mail\Plugin\Mail;
+
+use Drupal;
+use Exception;
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Mail\Plugin\Mail\PhpMail;
+
+/**
+ * Modify the drupal mail system to use theme template and replace tokens when sending emails.
+ *
+ * @Mail(
+ *   id = "falcon_mail_system",
+ *   label = @Translation("Falcon Mailer"),
+ *   description = @Translation("Provides formatter that can wrap emails into HTML templates and replace tokens.")
+ * )
+ */
+class FalconMailSystem extends PhpMail {
+
+  /**
+   * Concatenate and wrap the e-mail body for either HTML emails.
+   *
+   * @param array $message
+   *   A message array.
+   *
+   * @return array
+   *   The formatted $message.
+   */
+  public function format(array $message) {
+    $message['subject'] = !empty($message['subject']) ? $message['subject'] :
+      (isset($message['params']['subject']) ? $message['params']['subject'] : '');
+    $message['body'] = !empty($message['body']) ? $message['body'] :
+      (isset($message['params']['body']) ? $message['params']['body'] : []);
+    $message['from'] = !empty($message['from']) ? $message['from'] :
+      (isset($message['params']['from']) ? $message['params']['from'] : Drupal::config('system.site')->get('mail'));
```

Comment:
I'm very curious why PHP code sniffer doesn't complain because of `Drupal::` usage inside classes. Usually it asks to use dependency injection instead. Do you know why it's not the case here?

Reply:
Probably because i use "use Drupal;" in top of file (not sure). Should i rewrite it on dependency injection ?